cincinnati/src/lib.rs

```diff
@@ -187,26 +188,31 @@ impl Graph {
 
     /// Add a transition (edge) from `source` to `target`.
     ///
-    /// Fails with the `WoulcCycle` error if the new edge would lead to a cycle.
-    pub fn add_edge(&mut self, from: &ReleaseId, to: &ReleaseId) -> Result<(), Error> {
+    /// Fails with the `WouldCycle` error if the new edge would lead to a cycle.
+    pub fn add_edge(&mut self, from: &ReleaseId, to: &ReleaseId) -> Result<EdgeIndex, Error> {
+        let from_release: String = self.find_by_releaseid(from)?.version().to_string();
+        let to_release: String = self.find_by_releaseid(to)?.version().to_string();
         if self.dag.find_edge(from.0, to.0).is_some() {
             return Err(Error::from(errors::EdgeAlreadyExists {
-                from: self.find_by_releaseid(from)?.version().to_string(),
-                to: self.find_by_releaseid(to)?.version().to_string(),
+                from: from_release,
+                to: to_release,
             }));
         }
 
         self.dag
             .add_edge(from.0, to.0, Empty {})
-            .map(|_| ())
             .map_err(Into::into)
     }
 
     /// Add edges for all given key/value pairs of releases.
-    pub fn add_edges(&mut self, indices: HashMap<ReleaseId, ReleaseId>) -> Result<(), Error> {
+    pub fn add_edges(
+        &mut self,
+        indices: HashMap<ReleaseId, ReleaseId>,
+    ) -> Result<Vec<EdgeIndex>, Error> {
```

Comment:
I'm not sure if this is useful. Iterating a HashMap isn't ordered, and so it's impossible for the caller to know which key/value pair resulted in which edge index. You could return the releaseids with the edgeindex if you think this is useful. E.g.

```suggestion
    ) -> Result<Vec<(EdgeIndex, ReleaseId, ReleaseId)>, Error> {
```

Reply:
iiuc this is not even used anywhere

Reply:
If that's the case we can just leave this function alone for now.